pallet/src/lib.rs

```diff
@@ -308,7 +329,11 @@ pub mod pallet {
             let result = result::from_vm_result::<T>(vm_result)?;
 
             // Emit an event.
-            let signers = signature_handler.into_signer_accounts()?;
+            let mut signers = signature_handler.into_signer_accounts()?;
+            if signers.is_empty() {
+                // Signer list can be empty in zero-signer scripts, so append here the user at least.
+                signers.push(who);
```

Comment:
That seems a bit strange since this should already be done before when `sign_script` is called. Respectively, when calling `sign_script` this is done internally... :thinking:

Reply:
This is only for the case when we execute a script without a signer.
Without this logic here, the `who` variable would be empty, and then ExecuteEvent would look very strange when it would say it was executed by "None"